jans-cedarling/cedarling/src/common/policy_store.rs

```diff
@@ -5,155 +5,430 @@
-                include_str!("../../../test_files/policy-store_policy_err_broken_policy.json");
+    /// Tests that a valid version string with 'v' prefix is accepted.
+    #[test]
+    fn test_valid_version_with_v() {
+        let valid_version_with_v = "v1.2.3".to_string();
+        assert!(check_cedar_version(serde_json::Value::String(valid_version_with_v)).is_ok());
+    }
 
-            let policy_result = serde_json::from_str::<PolicyStoreMap>(POLICY_STORE_RAW);
-            let err_msg = policy_result.unwrap_err().to_string();
-            assert_eq!(err_msg,"unable to decode policy with id: 840da5d85403f35ea76519ed1a18a33989f855bf1cf8, error: unable to decode policy_content from human readable format: unexpected token `)` at line 15 column 1")
-        }
+    /// Tests that an invalid version format is rejected.
+    #[test]
+    fn test_invalid_version_format() {
+        let invalid_version = "1.2".to_string();
+        assert!(check_cedar_version(serde_json::Value::String(invalid_version)).is_err());
+    }
+
+    /// Tests that an invalid version part (non-numeric) is rejected.
+    #[test]
+    fn test_invalid_version_part() {
+        let invalid_version = "1.two.3".to_string();
+        assert!(check_cedar_version(serde_json::Value::String(invalid_version)).is_err());
+    }
+
+    /// Tests that an invalid version format with 'v' prefix is rejected.
+    #[test]
+    fn test_invalid_version_format_with_v() {
+        let invalid_version_with_v = "v1.2".to_string();
+        assert!(check_cedar_version(serde_json::Value::String(invalid_version_with_v)).is_err());
+    }
+
+    /// Tests that an error is returned for multiple role mappings in token metadata.
+    #[test]
+    fn test_invalid_multiple_role_mappings_in_token_metadata() {
+        let invalid_token_metadata = r#"[ 
+            { "type": "Access", "person_id": "aud" }, 
+            { "type": "Id", "person_id": "sub", "role_mapping": "role" }, 
+            { "type": "userinfo", "person_id": "email", "role_mapping": "role" } 
+        ]"#;
+
+        let token_metadata_value: serde_json::Value =
+            serde_json::from_str(invalid_token_metadata).expect("Failed to parse JSON string");
+
+        assert!(
+            check_token_metadata(token_metadata_value).is_err(),
+            "expected an error for multiple role mappings"
+        );
+    }
+
+    /// Tests successful parsing of role mappings in token metadata.
+    #[test]
+    fn test_successful_parsing_of_role_mappings() {
+        let valid_token_metadata = r#"[ 
+            { "type": "Access", "person_id": "aud" }, 
+            { "type": "Id", "person_id": "sub", "role_mapping": "role" }, 
+            { "type": "userinfo", "person_id": "email" } 
+        ]"#;
+
+        let token_metadata_value: serde_json::Value =
+            serde_json::from_str(valid_token_metadata).expect("Failed to parse JSON string");
+
+        assert!(
+            check_token_metadata(token_metadata_value).is_ok(),
```

Comment:
Consider adding a validation check to ensure that the parsed data is accurate and matches the expected structure defined in the JSON.

Reply:
the `check_token_metadata` already checks if the data is correct for Cedarling's use-case.

If you're asking to check if the JSON is getting parsed into the struct correctly, aren't i making unit-tests for `serde_json` at that point? i think we should just assume that `serde_json` parses correctly...

Reply:
In case of unit testing, it is great not just check if result is OK, but also check result. For some case it can be missed.

I leave this question up to you. You can click `Resolve conversation`